src/Core/src/ViewExtensions.cs

```diff
@@ -27,6 +28,47 @@ namespace Microsoft.Maui
 {
 	public static partial class ViewExtensions
 	{
+
+#pragma warning disable RS0016 // Add public types and members to the declared API
+		public static void DisconnectHandlers(this IView view)
+#pragma warning restore RS0016 // Add public types and members to the declared API
+		{
+			// For our first go here
+			// My thinking is to build a flat list of all views in the tree
+			// And then iterate down the list disconnecting handlers.
+			// This gives me a stable list of views to call disconnecthandler on
+			// I'm assuming as this PR evolves we'll probably add some interfaces
+			// that allow handlers to manage their own children and disconnect flow
+			List<IView> _flatList = new List<IView>();			
+			BuildFlatList(view, _flatList);
```

Comment:
This is OK for now, but I am wondering about the perf issues with creating this list. Maybe we can use a recursive function that first disconnects the children and then itself?

```cs
void DoDisconnect(IVisualTreeElement element)
{
    if (!element.ShouldDisconnectLogicHere)
        return;

    foreach (var child in vte.GetVisualChildren())
    {
        DoDisconnect(child);
    }
    element.Handler?.DisconnectHandler();
}
```

Reply:
yea, my concern is losing track of children because of the disconnect process, or the size of visual children changing for some reason. I use the flat list to try to avoid as many side effects as possible.

- Here's every element in the visual tree
- I am now going to call disconnect on all of them